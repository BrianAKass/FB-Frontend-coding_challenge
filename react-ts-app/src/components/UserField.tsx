import React, { ReactElement, useState } from "react";
import { setTimeout } from "timers";

import "../styles/UserField.css";

export interface Props {
  title?: string;
  first?: string;
  last?: string;
  picture?: string;
  email?: string;
  handleOpen: Function;
  handleModal: Function;
}

export default function UserField({
  title,
  first,
  last,
  picture,
  email,
  handleOpen,
  handleModal,
}: Props): ReactElement {
  const [clicked, setClicked] = useState(false);
  const handleClick = (title: any, first: any, last: any, email: any) => {
    handleModal(title, first, last, email);
    handleOpen();
  };
  const handleCopy = () => {
    navigator.clipboard.writeText(email!);
    setClicked(true);
    setTimeout(() => {
      setClicked(false);
    }, 500);
  };
  return (
    <div className="twoToneCard">
      <div className="top" />
      <img src={picture} alt="profile-pic" className="profile" />
      <h2>
        <span>{title}</span>
        {first} {last}
      </h2>
      <div className="bottom">
        <span
          className={!clicked ? "email" : "copied"}
          data-toggle="tooltip"
          data-placement="left"
          title={`Copy Email: ${email} to clipboard`}
          onClick={() => {
            handleCopy();
          }}
        >
          <i className="fas fa-envelope" /> {!clicked ? "Email" : "Copied!"}
        </span>
        <span
          className="edit"
          data-toggle="tooltip"
          data-placement="right"
          title="Edit User Data"
          onClick={() => handleClick(title, first, last, email)}
        >
          <i className="fas fa-edit" /> Edit
        </span>
      </div>
    </div>
  );
}
