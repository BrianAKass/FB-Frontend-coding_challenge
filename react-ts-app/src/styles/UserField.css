.twoToneCard {
  position: relative;
  overflow: hidden;
  min-width: 230px;
  max-width: 315px;
  width: 100%;
  color: #ffffff;
  text-align: left;
  line-height: 1.4em;
  border-radius: 5%;
  box-shadow: 0 0 0 1px #ffffffa6;
  margin: 1rem;
  animation: cardFade 1s;
}

@keyframes cardFade {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

* {
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  -webkit-transition: all 0.25s ease;
  transition: all 0.25s ease;
}
div .top {
  height: 10rem;
  background: #555555;
  vertical-align: top;
  opacity: 0.85;
  box-shadow: 0 0 0 2px #121212;
}
.twoToneCard .bottom {
  width: 100%;
  background-color: #141414;
  padding: 25px;
  position: relative;
}

.twoToneCard .bottom:before {
  position: absolute;
  content: "";
  bottom: 100%;
  left: 0;
  width: 0;
  height: 0;
  border-style: solid;
  border-width: 55px 0 0 400px;
  border-color: transparent transparent transparent #141414;
}
.twoToneCard .bottom span {
  opacity: 0.65;
  padding: 5px;
  border: 1px solid #ffffff;
  border-radius: 5px;
  color: #ffffff;
  font-size: 1em;
  text-transform: uppercase;
  margin: 10px 0;
  display: inline-block;
  width: 47%;
  text-align: center;
  text-decoration: none;
  font-weight: 600;
  letter-spacing: 1px;
}
.twoToneCard .profile {
  position: absolute;
  bottom: 40%;
  left: 25px;
  margin: 1rem 0;
  max-height: 100px;
  border-radius: 50%;
  opacity: 1;
  z-index: 2;
  box-shadow: 0 0 0 1px #f4f4f4;
}

.twoToneCard div .email {
  margin-right: 4%;
  border-color: #37afff;
  color: #37afff;
}
.twoToneCard div .copied {
  opacity: 1;
  margin-right: 4%;
  color: #141414;
  background-color: #37afff;
}
.twoToneCard div .edit {
  border-color: #ffffff;
  color: #ffffff;
}

.twoToneCard div .email:hover {
  opacity: 1;
}
.twoToneCard div .edit:hover {
  opacity: 1;
}

.twoToneCard h2 {
  position: absolute;
  bottom: 50%;
  right: 0px;
  width: 10rem;
  z-index: 2;
  padding: 0 25px;
  font-weight: 300;
  color: #ffffff;
}
.twoToneCard h2 span {
  display: block;
  font-size: 0.5em;
  color: #ffffff;
}

@media (max-width: 500px) {
  .twoToneCard h2 {
    padding: 0 25px 0 50px;
  }
}
.twoToneCard p {
  margin: 0 0 10px;
  font-size: 0.8em;
  letter-spacing: 1px;
  opacity: 0.8;
}
