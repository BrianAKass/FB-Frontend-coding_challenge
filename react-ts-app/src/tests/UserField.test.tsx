import { render } from "@testing-library/react";
import UserField, { Props } from "../components/UserField";
import { testPeople } from "./Testpeople";

const defaultProps = {
  title: "Mr",
  first: "Test",
  last: "123",
  picture: "test.jpg",
  email: "test@test.com",
  handleOpen() {
    return;
  },
  handleModal() {
    return;
  },
};

describe("<UserField/>", () => {
  it("UserField rendered", () => {
    const userField = render(<UserField {...defaultProps} />);
    expect(userField).toMatchSnapshot();
  });
});
