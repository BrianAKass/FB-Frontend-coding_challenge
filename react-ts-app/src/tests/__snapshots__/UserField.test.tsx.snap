// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserField/> UserField rendered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="twoToneCard"
      >
        <div
          class="top"
        />
        <img
          alt="profile-pic"
          class="profile"
          src="test.jpg"
        />
        <h2>
          <span>
            Mr
          </span>
          Test
           
          123
        </h2>
        <div
          class="bottom"
        >
          <span
            class="email"
            data-placement="left"
            data-toggle="tooltip"
            title="Copy Email: test@test.com to clipboard"
          >
            <i
              class="fas fa-envelope"
            />
             
            Email
          </span>
          <span
            class="edit"
            data-placement="right"
            data-toggle="tooltip"
            title="Edit User Data"
          >
            <i
              class="fas fa-edit"
            />
             Edit
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="twoToneCard"
    >
      <div
        class="top"
      />
      <img
        alt="profile-pic"
        class="profile"
        src="test.jpg"
      />
      <h2>
        <span>
          Mr
        </span>
        Test
         
        123
      </h2>
      <div
        class="bottom"
      >
        <span
          class="email"
          data-placement="left"
          data-toggle="tooltip"
          title="Copy Email: test@test.com to clipboard"
        >
          <i
            class="fas fa-envelope"
          />
           
          Email
        </span>
        <span
          class="edit"
          data-placement="right"
          data-toggle="tooltip"
          title="Edit User Data"
        >
          <i
            class="fas fa-edit"
          />
           Edit
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
